include/RAJA/LegacyCompatibility.hxx

```diff
@@ -0,0 +1,215 @@
+#ifndef KRIPKE_LEGAGY_COMPATIBILITY_HXX
+#define KRIPKE_LEGAGY_COMPATIBILITY_HXX 1
+
+static_assert(__cplusplus >= 201103L, "C++ standards below 2011 are not supported");
+
+#include <type_traits>
+#include <utility>
+#include <iostream>
+#include <functional>
+#include <cstdint>
+
+namespace VarOps {
```

Comment:
I would further suggest for clarity that we remove all the vowels from our namespace labels…

From: David Poliakoff <notifications@github.com<mailto:notifications@github.com>>
Reply-To: LLNL/RAJA <reply@reply.github.com<mailto:reply@reply.github.com>>
Date: Friday, May 6, 2016 at 9:20 AM
To: LLNL/RAJA <RAJA@noreply.github.com<mailto:RAJA@noreply.github.com>>
Cc: Rich Hornung <hornung1@llnl.gov<mailto:hornung1@llnl.gov>>, Mention <mention@noreply.github.com<mailto:mention@noreply.github.com>>
Subject: Re: [LLNL/RAJA] Codegen must die (#7)


In include/RAJA/LegacyCompatibility.hxx<https://github.com/LLNL/RAJA/pull/7#discussion_r62352845>:

> @@ -0,0 +1,215 @@
> +#ifndef KRIPKE_LEGAGY_COMPATIBILITY_HXX
> +#define KRIPKE_LEGAGY_COMPATIBILITY_HXX 1
> +
> +static_assert(__cplusplus >= 201103L, "C++ standards below 2011 are not supported");
> +
> +#include <type_traits>
> +#include <utility>
> +#include <iostream>
> +#include <functional>
> +#include <cstdint>
> +
> +namespace VarOps {


I feel that impl has a certain association in c++11, I think it's being used correctly here, so I'd just use it, personally. Though if we're going with facetious suggestions, I suggest raja::openmp::kokkos::cuda::openacc::chapel::agency::fiddling_with_magnets

—
You are receiving this because you were mentioned.
Reply to this email directly or view it on GitHub<https://github.com/LLNL/RAJA/pull/7/files/2e4346c272a4383ea030c6365f55a27ac4181f10#r62352845>